corehq/form_processor/backends/couch/processor.py

```diff
@@ -147,39 +147,44 @@ def get_cases_from_forms(case_db, xforms):
         return touched_cases
 
     @staticmethod
-    def hard_rebuild_case(domain, case_id, detail, save=True):
-        try:
-            case = CommCareCase.get(case_id)
-            assert case.domain == domain
-            found = True
-        except CaseNotFound:
+    def hard_rebuild_case(domain, case_id, detail, save=True, lock=True):
+        case, lock_obj = FormProcessorCouch.get_case_with_lock(case_id, lock=lock)
+        found = bool(case)
+        if found:
             case = CommCareCase()
             case.case_id = case_id
             case.domain = domain
-            found = False
+            if lock:
+                lock_obj = CommCareCase.get_obj_lock_by_id(case_id)
```

Comment:
This is reassigning a name that pointed to a different lock that [has already been acquired](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/dimagi/utils/couch/__init__.py#L180-L182). Can that lock be used instead of getting a new one here? If not, the other lock should be released before the new one is assigned to `lock_obj` here.

Reply:
Apologies, bug in my logic. But if a case isn't found then `lock_obj` will be None so we need to get a new lock and acquire it.